Convert README from reStructuredText to Markdown

Srijani-Das07 opened the pull request:
This pull request addresses issue #38.

It converts the existing README from reStructuredText (.rst) to GitHub-flavored Markdown (.md), improving readability and rendering consistency on GitHub. 

The Markdown version has been verified to preserve the original formatting and content. The original `README.rst` has been removed as part of this update.

- [x] I have read the **CONTRIBUTING** file
- [x] I will abide by the code of conduct

freakboy3742 commented on README.md:
```diff
@@ -0,0 +1,101 @@
+---
+title: Briefcase Linux System App Template
+---
```
I'm not sure what the intention is with this block, but it's not required.

freakboy3742 reviewed (changes requested):
Thanks for the PR; however, there's a couple of little details that need attention.

1. If you click on "view file" on the PR, there should be no difference between the rendering of the markdown version and the ReST version. This definitely isn't the case at present - page headings and title aren't at the same HTML level.
2. This template repository doesn't use change files, so there's no need for the changenote.

Srijani-Das07 commented:
Thank you for the feedback. I've removed the changenote and updated the title and page heading sizes to match the original rendering. Please let me know if there’s anything else you'd like me to change.

freakboy3742 reviewed (approved):
Awesome - thanks for the update!
